metalarchives_importer.user.js

```diff
@@ -0,0 +1,175 @@
+    }
+    return rdata;
+}
+
+function retrieveReleaseInfo() {
+    var release = new Object();
+    release.artist_credit = new Array();
+    release.artist_credit.push({
+        artist_name: $('h2.band_name').text(),
+        credited_name: $('h2.band_name').text()
+    });
+    release.title = $('h1.album_name').text();
+
+    var rdata = getGenericalData();
+
+    release = setreleasedate(release, rdata["Release date"]);
+    //todo add case for multiple labels if such a case exist
+    release.labels = new Array();
+    release.labels.push({
+        name: (rdata["Label"] == "Independent" ? "[no label]" : rdata["Label"]),
+        catno: (rdata["Catalog ID"] == "N/A" ? "" : rdata["Catalog ID"])
+    });
+
+    release.type = ReleaseTypes[rdata["Type"]][0];
+    release.secondary_types = new Array(ReleaseTypes[rdata["Type"]][1]);
+    release.status = 'official';
+
+    rdata.hasOwnProperty("Version desc.") && rdata["Version desc."].indexOf("Digipak") != -1 ? release.packaging = "Digipak" : release.packaging = "";
+
+    var identifiers = $("#album_tabs_notes > div:nth-child(2)").find("p:not([class])").contents();
+    for (var j = 0; j < identifiers.length; j++) {
+        if (identifiers[j].textContent.indexOf("Barcode:") != -1) {
+            release.barcode = $.trim(identifiers[j].textContent.substring(8));
+            break;
+        }
+    }
+
+    // Release URL
+    // URLs
+    var link_type = MBImport.URL_TYPES;
+    release.urls = new Array();
+    release.urls.push({
+        url: window.location.href,
+        link_type: link_type.other_databases
+    });
+
+    var releaseNumber = 1;
+    release.discs = new Array();
+    release.discs.push(new Object());
+    release.discs[releaseNumber - 1].tracks = new Array();
+    release.discs[releaseNumber - 1].format = ReleaseFormat[rdata["Format"]];
+    var tracksline = $('table.table_lyrics tr.even,table.table_lyrics tr.odd');
+    var trackslinelength = tracksline.length;
+
+    tracksline.each(function(index, element) {
+        var trackNumber = $.trim(element.children[0].textContent).replace('.', "");
+        if (trackNumber == "1" && trackNumber != index + 1) {
+            releaseNumber++;
+            release.discs.push(new Object());
+            release.discs[releaseNumber - 1].tracks = new Array();
+            release.discs[releaseNumber - 1].format = ReleaseFormat[rdata["Format"]];
+        }
+
+        var track = {
+            'number': trackNumber,
+            'title': element.children[1].textContent,
+            'duration': element.children[2].textContent,
+            'artist_credit': [release.artist_credit]
+        };
+        release.discs[releaseNumber - 1].tracks.push(track);
+    });
+    return release;
+}
+
+// Insert links in page
+function insertLink(release) {
+
+    var edit_note = 'Imported from ' + window.location.href;
+    var parameters = MBImport.buildFormParameters(release, edit_note);
+    var innerHTML = MBImport.buildFormHTML(parameters);
```

Comment:
You can also add the Search button, see https://github.com/murdos/musicbrainz-userscripts/blob/master/bandcamp_importer.user.js#L212